pytorch_lightning/trainer/training_tricks.py

```diff
@@ -80,3 +85,196 @@ def configure_accumulated_gradients(self, accumulate_grad_batches):
             self.accumulation_scheduler = GradientAccumulationScheduler(schedule)
         else:
             raise TypeError("Gradient accumulation supports only int and dict types")
+
+    def scale_batch_size(self,
+                         model: LightningModule,
+                         mode: str = 'power',
```

Comment:
How would one switch between these modes as a user?

Reply:
I idea is that the if the user wants to control this, they can invoke the method themself. They could do something like:
```
trainer = Trainer()
new_batch_size = trainer.scale_batch_size(model, ...) # control arguments yourself
model.hparams.batch_size = new_batch_size
trainer.fit(model)
```
however, most people probably just want to set the trainer flag and not care about what happens "under the hood".

Reply:
agreed with the flag. i think this needs to be consistent with our other apis

```python
Trainer(auto_scale_batch_size=True) # uses power
Trainer(auto_scale_batch_size='binsearch|power')
```
and make it a requirement that this will work on self.hparams.batch_size?